bin/src/hickory-dns.rs

```diff
@@ -284,6 +309,19 @@ async fn async_run(args: Cli) -> Result<(), String> {
         .map_err(|e| format!("failed to register signal handler: {e}"))?;
 
     let mut catalog: Catalog = Catalog::new();
+    catalog.set_nsid(args.nsid);
```

Comment:
Missed this the first time, but would it perhaps make more sense to handle NSID at the `ServerFuture` level instead of the `Catalog` level?

Reply:
It's the `Catalog` that looks at the request EDNS, decides if the version is compatible, and constructs the initial response EDNS. I thought that made it a pretty natural place to handle NSID.

Reply:
If you think it's easy, that might be nice.